### PR TITLE
AdminMenu Navigation (returnUrl)

### LINKED ISSUE #8169: Edit Widgets and Layers UI navigation
A PR will be provided with suggested fixes for the 2 issues below.

- **In the screen for adding/editing Layers there is no cancel button to return to the main form without saving**.

image

Expected behaviour:
Offer navigation option with a cancel button.
image

- **When performing certain actions in the Edit Widgets and Layers UI it navigates back to Index action without menu id**.

For example:

Add a widget to a zone:
image

When returning after creating the widget the url don't contain the admin menu id.
image

If the user navigates to another feature:
image

And returns by navigating back (Back button or keyboard shortcut in the browser) the menu highlighting is out of sync. In the example Placement is still selected in the menu, but the active screen is Edit Widgets and Layers.
image

Expected behaviour:
The menu should stay in sync with the active feature screen.

### PR DESCRIPTION
**Pull request by microposmp**
Fixes #8169  

- Added cancel button to layers UI for navigating back to main feature screen.
- Use FullRequestPath as return url for all actions.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Layers/Controllers/AdminController.cs [hunk 1]
```diff
@@ -104,18 +104,19 @@ public async Task<IActionResult> Index(LayersIndexViewModel model)
...
 
-        public async Task<IActionResult> Create()
+        public async Task<IActionResult> Create(string returnUrl = null)
         {
             if (!await _authorizationService.AuthorizeAsync(User, Permissions.ManageLayers))
             {
                 return Forbid();
             }
 
+            ViewData["ReturnUrl"] = returnUrl;
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Layers/Controllers/AdminController.cs:114]: Is it necessary?

Comment (microposmp) [reply src/OrchardCore.Modules/OrchardCore.Layers/Controllers/AdminController.cs]: No, it wasn't necessary. Changed to not use ViewData.

**Comment by microposmp** [comment]
There are some other modules that use ViewData for passing the returnUrl:

- ContentTypes (Adding/Editing fields)
- Placements
- Templates

The below modules didn't handle the returnUrl (can get out of sync with menu when navigating):

- Roles
- Shortcodes
- Users

I've prepared changes for the above modules.

@Skrypt Is it ok to add changes for the above modules to this PR (and rename it) or should I create a new PR?

**Comment by Skrypt** [comment]
If it's related changes you can push your commits on top of this PR.

**Comment by Skrypt** [comment]
I've not tested yet. But I hope you are sure we needs this everywhere. Is there not a more generic way to do this while you are working on it?

**Comment by Piedone** [comment]
Is this something you'd like to revisit any time soon @microposmp (given that this is a draft PR) or should we close?

**Comment by microposmp** [comment]
I'll not do anything soon. We can close it.

**Pull request closed without merging**